### ISSUE TITLE
Generic/NestingLevel: support anonymous functions

### ISSUE THREAD

**Issue by rodrigoprimo** [PHPCSStandards/PHP_CodeSniffer#761]
## Is your feature request related to a problem?

Similar to #742, while working on improving code coverage for `Generic.Metrics.NestingLevel`, I noticed that the sniff doesn't check the nesting level of anonymous functions, and I believe it should.

- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [ ] I intend to create a pull request to implement this feature.

**Comment by jrfnl** [comment]
Makes sense to me (and once support for PHP 8.4 property hooks gets added, those should probably also be checked).

Note: checking closures declared via arrow functions probably doesn't make sense as those can't contain more than one statement.
